ui/imports/shared/stores/TokenBalanceHistoryStore.qml

```diff
@@ -39,37 +56,42 @@ ChartStoreBase {
             break;
             default:
                 console.warn("Invalid or unsupported time range")
+                return
             break;
         }
-        root.newDataReady(timeRange)
+
+        d.address = address
+        d.tokenSymbol = tokenSymbol
+        d.currencySymbol = currencySymbol
+
+        root.newDataReady(address, tokenSymbol, currencySymbol, timeRange)
     }
 
-    /// \arg timeRange: of type ChartStoreBase.TimeRange
-    function resetData(timeRange) {
-        root.setData(timeRange, [], [])
+    function resetAllData(address, tokenSymbol, currencySymbol) {
+        for (let tR = ChartStoreBase.TimeRange.Weekly; tR <= ChartStoreBase.TimeRange.All; tR++) {
+            root.setData(address, tokenSymbol, currencySymbol, tR, [], [])
+        }
     }
 
     Connections {
         target: walletSectionAllTokens
+
         function onTokenBalanceHistoryDataReady(balanceHistory: string) {
-           // chainId, address, symbol, timeInterval
+            // chainIds, address, tokenSymbol, currencySymbol, timeInterval
             let response = JSON.parse(balanceHistory)
             if (response === null) {
```

Comment:
JSON.parse will return null if balanceHistory == "null"
```
JSON.parse("null"); // null
```
I think this code is meant to check if balanceHistory is a valid Json and might be worth adding `try...catch`

Reply:
The object is already checked [here](https://github.com/status-im/status-desktop/blob/e4ab7165669aed71d7de11b05f417343d6ef363f/src/app_service/service/token/service.nim#L238) and build in NIM and status-go, so no exception or nil value expected. I improved the error handling, though; thanks for the heads up.